webapp/src/main/java/life/qbic/views/register/UserRegistrationHandler.java

```diff
@@ -90,9 +79,30 @@ public void onSuccess() {
 
   @Override
   public void onFailure(String reason) {
-    // Display error to the user
-    // Stub output:
-    System.out.println(reason);
-    userRegistrationLayout.alreadyUsedEmailMessage.setVisible(true);
+    handleRegistrationFailure(reason);
+  }
+
+  private void handleRegistrationFailure(String reason) {
+    switch (reason) {
+      case "Full Name shorter than 1 character." -> {
+        userRegistrationLayout.fullName.setInvalid(true);
+        userRegistrationLayout.fullName.setErrorMessage(reason);
+      }
+      case "Invalid email address format." -> {
+        userRegistrationLayout.email.setInvalid(true);
+        userRegistrationLayout.email.setErrorMessage(reason);
+      }
+      case "User with email address already exists." -> {
+        userRegistrationLayout.alreadyUsedEmailMessage.setVisible(true);
+        userRegistrationLayout.email.setInvalid(true);
+        userRegistrationLayout.email.setErrorMessage(reason);
+      }
+      case "Unexpected error occurred." -> userRegistrationLayout.errorMessage.setVisible(true);
+      case "Password shorter than 8 characters." -> {
+        userRegistrationLayout.passwordTooShortMessage.setVisible(true);
+        userRegistrationLayout.password.setInvalid(true);
+        userRegistrationLayout.password.setErrorMessage(reason);
+      }
```

Comment:
default case is missing. we could show an unexpected error or so :)

Reply:
Short remark, feel free to ignore:
Here an enum of possible reasons could make sense, you would not need a default then.

Reply:
In my dreamworld i'd define the reason set in the domain logic in an enum which can also be used in the handler for checking. But this would violate clean architecture 🤷‍♂️